src/Voron/StorageEnvironment.cs

```diff
@@ -883,72 +807,24 @@ public long PossibleOldestReadTransaction(LowLevelTransaction tx)
             return result;
         }
 
-        internal ExitWriteLock PreventNewTransactions()
-        {
-            if (_txCreation.IsWriteLockHeld)
-                return default;
-
-            _txCreation.EnterWriteLock();
-            return new ExitWriteLock(_txCreation);
-        }
-
-        internal bool IsInPreventNewTransactionsMode => _txCreation.IsWriteLockHeld;
-
-        internal bool TryPreventNewTransactions(TimeSpan timeout, out IDisposable exitWriteLock)
-        {
-            if (_txCreation.TryEnterWriteLock(timeout))
-            {
-                exitWriteLock = new ExitWriteLock(_txCreation);
-                return true;
-            }
-
-            exitWriteLock = null;
-            return false;
-        }
-
-        public struct ExitWriteLock : IDisposable
-        {
-            private readonly ReaderWriterLockSlim _rwls;
-
-            public ExitWriteLock(ReaderWriterLockSlim rwls)
-            {
-                _rwls = rwls;
-            }
-
-            public void Dispose()
-            {
-                _rwls?.ExitWriteLock();
-            }
-        }
-
         public event Action<LowLevelTransaction> NewTransactionCreated;
-        public event Action<LowLevelTransaction> AfterCommitWhenNewTransactionsPrevented;
 
         internal void TransactionAfterCommit(LowLevelTransaction tx)
         {
             tx._forTestingPurposes?.ActionToCallOnTransactionAfterCommit?.Invoke();
 
             if (ActiveTransactions.Contains(tx) == false)
             {
-                if (tx.Committed && tx.FlushedToJournal)
+                if (tx.Committed && tx.FlushedToJournal >= 0)
                     ThrowCommittedAndFlushedTransactionNotFoundInActiveOnes(tx);
 
                 return;
             }
 
-            using (PreventNewTransactions())
-            {
-                if (tx.Committed && tx.FlushedToJournal)
-                    Interlocked.Exchange(ref _transactionsCounter, tx.Id);
-
-                State = tx.State;
-
-                Journal.Applicator.OnTransactionCommitted(tx);
-                ScratchBufferPool.UpdateCacheForPagerStatesOfAllScratches();
-                Journal.UpdateCacheForJournalSnapshots();
-
-                tx.OnAfterCommitWhenNewTransactionsPrevented();
-            }
+            if (!tx.Committed) 
```

Comment:
Can we change to:
```suggestion
            if (tx.Committed == false) 
```

for better readability?